File: hw2/RaceMonitor_CSharp/RaceData/Messages/DidNotFinishUpdate.cs
using System;

namespace RaceData.Messages
{
    public class DidNotFinishUpdate : AthleteUpdate
    {
        public DidNotFinishUpdate() : base(AthleteRaceStatus.DidNotFinish) { }

        public DidNotFinishUpdate(string[] properties) : base(AthleteRaceStatus.DidNotFinish, properties)
        {
            if (properties.Length != 3 || properties[0] != AthleteRaceStatus.DidNotFinish.ToString())
                throw new ApplicationException("Invalid properties");
        }

    }
}

File: hw2/RaceMonitor_Java/src/Messages/DidNotFinishUpdate.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Messages;

import Exceptions.ApplicationException;
import Racedata.AthleteRaceStatus;

/**
 *
 * @author Shimin
 * @author Stephen Clyde
 */
public class DidNotFinishUpdate extends AthleteUpdate {
    public DidNotFinishUpdate() {
        super(AthleteRaceStatus.DidNotFinish);
    }
    
    public DidNotFinishUpdate(String[] properties) throws ApplicationException {
        super (AthleteRaceStatus.DidNotFinish, properties);
        
        if (properties.length != 3 || !properties[0].equals(AthleteRaceStatus.DidNotFinish.toString()))
            throw new ApplicationException("Invalid properties");
    }
}
